src/layer/AbstractLayer.ts

```diff
@@ -44,41 +50,104 @@ export class AbstractLayer {
+    }
+
+    let tiles: Tile[] = [];
+    for (let i = 0; i < maxFindTilesRequests; i++) {
+      const { tiles: partialTiles, hasMore } = await this.findTiles(
+        bbox,
+        fromTime,
+        toTime,
+        tilesPerRequest,
+        i * tilesPerRequest,
+      );
+      tiles = tiles.concat(partialTiles);
+      if (!hasMore) {
+        break;
+      }
+      if (i + 1 === maxFindTilesRequests) {
+        throw new Error(
+          `Could not fetch all the tiles in [${maxFindTilesRequests}] requests for [${tilesPerRequest}] tiles`,
+        );
+      }
+    }
+
     tiles.sort((a, b) => a.sensingTime.getTime() - b.sensingTime.getTime());
     let orbitTimeMS = this.dataset.orbitTimeMinutes * 60 * 1000;
     let flyoverIntervals: FlyoverInterval[] = [];
 
-    let j = 0;
-    for (let i = 0; i < tiles.length; i++) {
-      if (i === 0) {
-        flyoverIntervals[j] = {
-          fromTime: tiles[i].sensingTime,
-          toTime: tiles[i].sensingTime,
+    let flyoverIndex = 0;
+    let currentFlyoverGeometry = null;
+    for (let tileIndex = 0; tileIndex < tiles.length; tileIndex++) {
+      if (tileIndex === 0) {
+        flyoverIntervals[flyoverIndex] = {
+          fromTime: tiles[tileIndex].sensingTime,
+          toTime: tiles[tileIndex].sensingTime,
+          coveragePercent: 0,
         };
+        currentFlyoverGeometry = tiles[tileIndex].geometry;
         continue;
       }
 
-      const prevDateMS = tiles[i - 1].sensingTime.getTime();
-      const currDateMS = tiles[i].sensingTime.getTime();
+      const prevDateMS = tiles[tileIndex - 1].sensingTime.getTime();
+      const currDateMS = tiles[tileIndex].sensingTime.getTime();
       const diffMS = Math.abs(prevDateMS - currDateMS);
 
       if (diffMS < orbitTimeMS) {
-        flyoverIntervals[j].toTime = tiles[i].sensingTime;
+        flyoverIntervals[flyoverIndex].toTime = tiles[tileIndex].sensingTime;
+        currentFlyoverGeometry = union(currentFlyoverGeometry, tiles[tileIndex].geometry);
       } else {
-        j++;
-        flyoverIntervals[j] = {
-          fromTime: tiles[i].sensingTime,
-          toTime: tiles[i].sensingTime,
+        flyoverIntervals[flyoverIndex].coveragePercent = this.calculateCoveragePercent(
+          bbox,
+          currentFlyoverGeometry,
+        );
+        flyoverIndex++;
+        flyoverIntervals[flyoverIndex] = {
+          fromTime: tiles[tileIndex].sensingTime,
+          toTime: tiles[tileIndex].sensingTime,
+          coveragePercent: 0,
         };
+        currentFlyoverGeometry = tiles[tileIndex].geometry;
       }
     }
+    if (flyoverIntervals.length > 0) {
+      flyoverIntervals[flyoverIndex].coveragePercent = this.calculateCoveragePercent(
+        bbox,
+        currentFlyoverGeometry,
+      );
+    }
     return flyoverIntervals;
   }
 
+  private calculateCoveragePercent(bbox: BBox, flyoverGeometry: Polygon | MultiPolygon): number {
+    const bboxGeometry: Polygon = {
```

Comment:
Feel free to resolve this and merge, but maybe we should try and add function to convert a bbox to a geometry as this is duplicated in a few spots already:
https://github.com/sentinel-hub/sentinelhub-js/blob/master/src/layer/AbstractSentinelHubV3Layer.ts#L163
https://github.com/sentinel-hub/sentinelhub-js/blob/master/src/layer/AbstractSentinelHubV1OrV2Layer.ts#L79

Reply:
Sounds like a good idea, yes. I would prefer todo it in a separate PR though, since this one is kind of needed asap... :)